src/kestrel_datasource_stixshifter/interface.py

```diff
@@ -109,14 +114,26 @@
 
 def check_module_availability(connector_name):
     try:
-        connector_module = importlib.import_module(
+        importlib.import_module(
             "stix_shifter_modules." + connector_name + ".entry_point"
         )
     except:
-        raise DataSourceError(
-            f'STIX shifter connector for "{connector_name}" is not installed',
-            "please install the corresponding STIX shifter connector Python package.",
+        package_name = "stix-shifter-modules-" + connector_name.replace("_", "-")
+        _logger.info(
+            f'installing missing stix-shifter connector "{connector_name}" '
+            f'with possible Python package "{package_name}"'
         )
+        subprocess.check_call([sys.executable, "-m", "pip", "install", package_name])
```

Comment:
I'm not convinced this is a good idea...

Reply:
This is a usability vs. security balance. It makes using a stix-shifter connector much more convenient (I couldn't find the mapping between connector name to package, so as many users who may not know stix-shifter well), but open a new attack surface where attackers may upload their own stix-shifter modules to pypi.

Reply:
Use PyPI to verify the stix-shifter connector package is genuine before install.